### PR TITLE
[WIP] Add ws event new

### PR DESCRIPTION
**Pull request by xVin**
targets #358

### CODE REVIEW

**Comment by xVin** [comment]
Is this what you had in mind?

**Comment by neithernut** [comment]
Yep. Pretty much.
ACK.

File: src/objects/message/event.c [hunk 1]
```diff
@@ -104,6 +104,28 @@ ws_event_init(
...
+    struct ws_event* ev = calloc(1, sizeof(*ev));
+
+    if (!ev) {
+        return NULL;
+    }
+
+    int res = ws_event_init(ev, name, ctx);
+
+    if (res != 0) {
+        return NULL;
```
Comment (matthiasbeyer) [inline src/objects/message/event.c:121]: Does not `free()` the allocated memory here.

Maybe you should provide a follow-up patch to fix this.

Comment (xVin) [inline src/objects/message/event.c:121]: True & done.

**Comment by neithernut** [comment]
Argh, I should stop merging PRs at 3am.

**Pull request merged**